Wrong order when applying plan

It tried to apply Policy before Site.

```
2020-09-24 07:41:27,382 [INFO] [appgate-operator/appgate-test] AppgatePlan Summary:
2020-09-24 07:41:27,383 [INFO] [appgate-operator/appgate-test] + <class 'appgate.openapi.parser.Policy'> simple_setup_POL [427d30cf-b4a9-4973-bd74-08df84d585e8]
2020-09-24 07:41:27,509 [ERROR] [aggpate-client] https://ec2-3-83-104-221.compute-1.amazonaws.com:444//admin//policies :: 422: {"id":"unprocessable entity","message":"Invalid reference to another entity"}
2020-09-24 07:41:27,509 [ERROR] [appgate-client] payload :: {'name': 'simple_setup_POL', 'expression': "if ( claims.user.username == 'bob' ) {\n return true;\n} \n\nreturn false;", 'tags': [], 'entitlements': ['9eb29d80-0bd2-47c9-b849-816962a81b19', 'd8f09f37-9401-4136-b9
69-95ebbb175c1c'], 'entitlementLinks': [], 'ringfenceRules': [], 'ringfenceRuleLinks': [], 'administrativeRoles': [], 'id': '427d30cf-b4a9-4973-bd74-08df84d585e8'}
2020-09-24 07:41:27,509 [ERROR] [aggpate-client] POST /admin//policies :: Expecting a response but we got empty data
2020-09-24 07:41:27,509 [INFO] [appgate-operator/appgate-test] + <class 'appgate.openapi.parser.Entitlement'> simple_setup_ENT_HTTP [d8f09f37-9401-4136-b969-95ebbb175c1c]
2020-09-24 07:41:27,632 [ERROR] [aggpate-client] https://ec2-3-83-104-221.compute-1.amazonaws.com:444//admin//entitlements :: 422: {"id":"unprocessable entity","message":"Validation failed.","errors":[{"field":"site","message":"does not exist"}]}
2020-09-24 07:41:27,633 [ERROR] [appgate-client] payload :: {'name': 'simple_setup_ENT_HTTP', 'site': '3d844467-8d82-42d2-946a-53b3d8ff11ec', 'tags': [], 'conditions': ['ee7b7e6f-e904-4b4f-a5ec-b3bef040643e'], 'actions': [{'subtype': 'tcp_up', 'action': 'allow', 'hosts':
['172.17.115.2'], 'ports': ['80', '443'], 'types': [], 'monitor': {}}], 'appShortcuts': [], 'appShortcutScripts': [], 'id': 'd8f09f37-9401-4136-b969-95ebbb175c1c'}
2020-09-24 07:41:27,633 [ERROR] [aggpate-client] POST /admin//entitlements :: Expecting a response but we got empty data
2020-09-24 07:41:27,633 [INFO] [appgate-operator/appgate-test] + <class 'appgate.openapi.parser.Entitlement'> simple_setup_ENT_PING [9eb29d80-0bd2-47c9-b849-816962a81b19]
2020-09-24 07:41:27,756 [ERROR] [aggpate-client] https://ec2-3-83-104-221.compute-1.amazonaws.com:444//admin//entitlements :: 422: {"id":"unprocessable entity","message":"Validation failed.","errors":[{"field":"site","message":"does not exist"}]}
2020-09-24 07:41:27,756 [ERROR] [appgate-client] payload :: {'name': 'simple_setup_ENT_PING', 'site': '3d844467-8d82-42d2-946a-53b3d8ff11ec', 'tags': ['was', 'here', 'operator', 'sdp'], 'conditions': ['ee7b7e6f-e904-4b4f-a5ec-b3bef040643e'], 'actions': [{'subtype': 'icmp_
up', 'action': 'allow', 'hosts': ['172.17.115.6', '172.17.115.2'], 'ports': [], 'types': ['0-255']}], 'appShortcuts': [], 'appShortcutScripts': [], 'id': '9eb29d80-0bd2-47c9-b849-816962a81b19'}
2020-09-24 07:41:27,757 [ERROR] [aggpate-client] POST /admin//entitlements :: Expecting a response but we got empty data
2020-09-24 07:41:27,757 [INFO] [appgate-operator/appgate-test] + <class 'appgate.openapi.parser.Appliance'> gateway-8b61286b-caf5-47df-8702-c1506a4afe3c-site1 [9f8cde7c-c231-4a86-b150-10a05c6e3648]
2020-09-24 07:41:27,887 [ERROR] [aggpate-client] https://ec2-3-83-104-221.compute-1.amazonaws.com:444//admin//appliances :: 422: {"id":"unprocessable entity","message":"Validation failed.","errors":[{"field":"site","message":"does not exist"}]}
2020-09-24 07:41:27,887 [ERROR] [appgate-client] payload :: {'name': 'gateway-8b61286b-caf5-47df-8702-c1506a4afe3c-site1', 'hostname': 'ec2-54-89-225-112.compute-1.amazonaws.com', 'clientInterface': {'hostname': 'ec2-54-89-225-112.compute-1.amazonaws.com', 'allowSources':
 [{'address': '0.0.0.0', 'netmask': 0}, {'address': '::', 'netmask': 0}]}, 'peerInterface': {'hostname': 'ec2-54-89-225-112.compute-1.amazonaws.com', 'allowSources': [{'address': '::', 'netmask': 0}, {'address': '0.0.0.0', 'netmask': 0}]}, 'networking': {'hosts': [], 'nic
s': [{'name': 'eth0', 'enabled': True, 'ipv4': {'dhcp': {'enabled': True, 'dns': True, 'routers': True, 'ntp': False, 'mtu': False}, 'static': []}, 'ipv6': {'dhcp': {'enabled': False, 'dns': True, 'ntp': False, 'mtu': False}, 'static': []}}], 'dnsServers': [], 'dnsDomains
': [], 'routes': []}, 'tags': [], 'site': '3d844467-8d82-42d2-946a-53b3d8ff11ec', 'ntp': {'servers': [{'hostname': '3.ubuntu.pool.ntp.org'}, {'hostname': '1.ubuntu.pool.ntp.org'}, {'hostname': '0.ubuntu.pool.ntp.org'}, {'hostname': '2.ubuntu.pool.ntp.org'}]}, 'sshServer':
 {'enabled': True, 'allowSources': [{'address': '0.0.0.0', 'netmask': 0}, {'address': '::', 'netmask': 0}]}, 'snmpServer': {'allowSources': []}, 'healthcheckServer': {'allowSources': [{'address': '::', 'netmask': 0}, {'address': '0.0.0.0', 'netmask': 0}]}, 'prometheusExpo
rter': {'allowSources': []}, 'ping': {'allowSources': [{'address': '::', 'netmask': 0}, {'address': '0.0.0.0', 'netmask': 0}]}, 'logServer': {}, 'controller': {}, 'gateway': {'enabled': True, 'vpn': {'allowDestinations': [{'nic': 'eth0'}]}}, 'logForwarder': {'tcpClients':
 [], 'awsKineses': [], 'sites': []}, 'connector': {'expressClients': [], 'advancedClients': []}, 'rsyslogDestinations': [], 'hostnameAliases': [], 'id': '9f8cde7c-c231-4a86-b150-10a05c6e3648'}
2020-09-24 07:41:27,887 [ERROR] [aggpate-client] POST /admin//appliances :: Expecting a response but we got empty data
2020-09-24 07:41:27,887 [INFO] [appgate-operator/appgate-test] + <class 'appgate.openapi.parser.Site'> simple_setup Site [3d844467-8d82-42d2-946a-53b3d8ff11ec
```

--- a/appgate/appgate.py
+++ b/appgate/appgate.py
@@ -304,7 +304,8 @@ async def main_loop(queue: Queue, ctx: Context, k8s_configmap_client: K8SConfigM
                                                             api_spec=ctx.api_spec,
                                                             appgate_client=appgate_client)
                                                         if appgate_client else {},
-                                                        k8s_configmap_client=k8s_configmap_client)
+                                                        k8s_configmap_client=k8s_configmap_client,
+                                                        api_spec=ctx.api_spec)
 
                     if appgate_client:
                         current_appgate_state = new_plan.appgate_state

--- a/appgate/state.py
+++ b/appgate/state.py
@@ -7,7 +7,7 @@ import datetime
 import time
 from functools import cached_property
 from pathlib import Path
-from typing import Set, Dict, Optional, Tuple, Literal, Any, Iterable, List, FrozenSet
+from typing import Set, Dict, Optional, Tuple, Literal, Any, Iterable, List, FrozenSet, Iterator
 
 import yaml
 from attr import attrib, attrs, evolve
@@ -342,14 +342,18 @@ class AppgatePlan:
     def needs_apply(self) -> bool:
         return any(v.needs_apply for v in self.entities_plan.values())
 
+    def ordered_entities_plan(self, api_spec: APISpec) -> Iterator[Tuple[str, Plan]]:
+        return map(lambda k: (k, self.entities_plan[k]), api_spec.entities_sorted)
+
 
 async def appgate_plan_apply(appgate_plan: AppgatePlan, namespace: str,
                              entity_clients: Dict[str, EntityClient],
-                             k8s_configmap_client: K8SConfigMapClient) -> AppgatePlan:
+                             k8s_configmap_client: K8SConfigMapClient,
+                             api_spec: APISpec) -> AppgatePlan:
     log.info('[appgate-operator/%s] AppgatePlan Summary:', namespace)
     entities_plan = {k: await plan_apply(v, namespace=namespace, entity_client=entity_clients.get(k),
                                          k8s_configmap_client=k8s_configmap_client)
-                     for k, v in appgate_plan.entities_plan.items()}
+                     for k, v in appgate_plan.ordered_entities_plan(api_spec)}
     return AppgatePlan(entities_plan=entities_plan)
 
 

